src/compiler/fs-watch/fs-watch-rebuild.ts

```diff
@@ -22,9 +22,14 @@ const unaryBasename = (filePath: string): string => basename(filePath);
  * Get the file extension for a path
  *
  * @param filePath a path
- * @returns the file extension (well, characters after the last `'.'`)
+ * @returns the file extension (well, characters after the last `'.'`) or
+ * `null` if no extension exists.
  */
-const getExt = (filePath: string): string => filePath.split('.').pop().toLowerCase();
+const getExt = (filePath: string): string | null => {
+  const fileParts = filePath.split('.');
+
+  return fileParts.length > 1 ? fileParts.pop()!.toLowerCase() : null;
```

Comment:
No action required - I think we can avoid having to use the bang operator if we break this down into 2 statements:
```suggestion
  const extension = fileParts.pop();
  return extension ? extension.toLowerCase() : null;
```
That said, I doubt the typings for `Array.pop` will change such that the bang operator we have here today will be unnecessary, so no action required